types/src/lib.rs

```diff
@@ -2618,6 +2618,28 @@ impl ModuleId {
         self.get(db).symbols.contains_key(name)
     }
 
+    pub fn import_symbol(self, db: &Database, name: &str) -> Option<Symbol> {
```

Comment:
Could you add some tests for this method? You can find an example (albeit a rather basic one) for `ModuleId::symbol` in `test_module_id_symbol`. To cover the cases such as `Symbol::Trait` you can use helper methods such as `types::test::new_trait`, which should take care of most of the boilerplate.

The test cases would be:

- A test that checks a `None` is returned when the symbol doesn't exist. This can just be done once, it doesn't need to be repeated for every symbol type.
- A test that covers the case `if self == module_id else` for every `Symbol` variant that produces a `ModuleId`
- A test that covers trying to import a type parameter

I added an example in e561b044f630e3f785edb758248cc5783bb03372 along with some helper code to not make this a total pain. Once you're done, please squash that commit into yours. Thanks! :smiley:

Reply:
Okay, this is now done, but I resorted to using `Trait::alloc()` and similar methods for other symbols because that was the easiest to set the proper module they are coming from. Do you want me to turn them into helpers like `new_class`/`new_module`?

Reply:
Also - I did it all in a single test function. Do you want me to split it into multiple test fns?